### ISSUE TITLE
dat clone from datproject.org errors without dir

### ISSUE THREAD

**Issue by joehand** [datproject/dat#792]
Errors if you do not specify directory:

```
❯ dat clone datproject.org/jhand/cli-demo
/Users/joe/node_modules/dat/node_modules/dat-node/lib/storage.js:50
      throw new Error('Invalid storage path')
      ^

Error: Invalid storage path
    at defaultStorage (/Users/joe/node_modules/dat/node_modules/dat-node/lib/storage.js:50:13)
    at createDat (/Users/joe/node_modules/dat/node_modules/dat-node/index.js:36:17)
    at runDat (/Users/joe/node_modules/dat/src/commands/clone.js:86:7)
    at /Users/joe/node_modules/dat/src/commands/clone.js:65:9
    at FSReqWrap.oncomplete (fs.js:123:15)
```

**Comment by joehand** [comment]
We should check this out after #845 is merged.

**Comment by TimothyStiles** [comment]
@joehand how should it behave if you don't specify a directory? Clone into cwd or clone into a new directory named after decoded key? I think I should be able to fix this.

**Comment by joehand** [comment]
> @joehand how should it behave if you don't specify a directory? Clone into cwd or clone into a new directory named after decoded key? I think I should be able to fix this.

It should clone into a new directory. Two options are: 

1. clone into key (consistent with other usage)
2. clone into dir named after url (e.g .`dat clone datproject.org/jhand/cli-demo` -> `./cli-demo`. But I'm not sure how that'd work in situations like `dat clone dat://beakerbrowser.com` (this currently clones into a directory with key - not sure why it succeeds where the other fails).

Option 2 sounds nicer we just have to think about those edge cases. May be worth doing the quick fix of option #1 first then doing number 2 (maybe along with https://github.com/datproject/dat/issues/793).

**Comment by buckle2000** [comment]
Maybe add `dat cat` like `ipfs cat`?
